apps/angular/1-projection/src/app/ui/list-item/list-item.component.ts

```diff
@@ -22,19 +19,7 @@ import { CardType } from '../../model/card.model';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ListItemComponent {
-  private teacherStore = inject(TeacherStore);
-  private studentStore = inject(StudentStore);
-
   readonly id = input.required<number>();
-  readonly name = input.required<string>();
-  readonly type = input.required<CardType>();
 
-  delete(id: number) {
-    const type = this.type();
-    if (type === CardType.TEACHER) {
-      this.teacherStore.deleteOne(id);
-    } else if (type === CardType.STUDENT) {
-      this.studentStore.deleteOne(id);
-    }
-  }
+  delete = output<number>();
```

Comment:
you don't need to output the id, you can just emit an event and the parent will passs the id to the function